pkg/epp/util/request/body.go

```diff
@@ -0,0 +1,70 @@
+package request
+
+import (
+	"fmt"
+
+	errutil "sigs.k8s.io/gateway-api-inference-extension/pkg/epp/util/error"
+)
+
+func ExtractPromptFromRequestBody(body map[string]interface{}) (string, error) {
+	_, ok := body["messages"]
+	if ok {
+		return extractPromptForChatCompletions(body)
+	}
+	return extractPromptForCompletions(body)
+}
```

Comment:
nit (very very nit):
can you call the helper functions in general names? 
for example, extractPromptField and extractPromptFromMessagesField?
yes, today we have chat and chat completions but AI world is moving fast and names/standards change while prompt field may stay.